documentation/specs/TextField.md

```diff
@@ -0,0 +1,307 @@
+
+export type TextFieldProps = AriaTextFieldProps & {
+  /**
+   * Underlying HTML input type
+   * @default 'text'
+   */
+  type?: InputType;
+  /**
+   * When set to `protected`, the underlying input is set to `password` and a right aligned
+   * clickable and focusable visibility icon is added. The net effect is to give a PasswordInput.
+   * @default 'standard'
+   */
+  variant?: TextFieldVariant;
+  /**
+   * TextField size affects the overall size of the input, but it also influences the size of
+   * leftAlignedIcon and rightAlignedIcon. For instance, with `sm` size, the associated leftAlignedIcon
+   * or rightAlignedIcon size maps to the Easy UI icon size of 'sm'.
+   * @default 'md'
+   */
+  size?: TextFieldSize;
+  /**
+   * Visually hides the label, but keeps it accessible.
+   * @default false
+   */
+  isLabelVisuallyHidden?: boolean;
+  /** Error text that appears below input */
+  errorText?: ReactNode;
+  /** Helper text that appears below input */
+  helperText?: ReactNode;
+  /** Left aligned icon */
+  leftAlignedIcon?: IconSymbol;
+  /** Right aligned icon */
+  rightAlignedIcon?: IconSymbol;
+};
+```
+
+### Anatomy
+
+The bulk of the `TextField` component behavior will be handled by React Aria's `useTextField` hook as it provides the behavior and accessibility implementation for a text field.
+
+```tsx
+export function TextField(props: TextFieldProps) {
+  let {
+    type = "text",
+    variant = "standard",
+    size = "md",
+    isLabelVisuallyHidden = false,
+    isDisabled = false,
+    isRequired = false,
+    validationState = "invalid",
+    leftAlignedIcon,
+    rightAlignedIcon,
+    label,
+    errorText,
+    helperText,
+    placeholder,
+    value,
+  } = props;
+  let ref = React.useRef(null);
+  let {
+    labelProps,
+    inputProps,
+    descriptionProps: helperTextProps,
+    errorMessageProps: errorTextProps,
+  } = useTextField(props, ref);
+
+  const showHelperText =
+    helperText && !errorText && validationState === "valid";
+  const showErrorText =
+    !showHelperText && errorText && validationState === "invalid";
+
+  const isPasswordInput = variant === "protected";
+
+  return (
+    <div>
+      <div>
+        <div>
+          {leftAlignedIcon && <Icon symbol={leftAlignedIcon} size={size} />}
+          <input
+            {...inputProps}
+            ref={ref}
+            type={type}
+            value={value}
+            required={isRequired}
+            disabled={isDisabled}
+            placeholder={placeholder}
+          />
+          {rightAlignedIcon && <Icon symbol={rightAlignedIcon} size={size} />}
+        </div>
+        <label {...labelProps}>
+          <Text>{label}</Text>
+        </label>
+      </div>
+      {showHelperText && (
+        <div {...helperTextProps}>
+          <Text>{helperText}</Text>
+        </div>
+      )}
+      {showErrorText && (
+        <div {...errorTextProps}>
+          <Text>{errorText}</Text>
+        </div>
+      )}
+    </div>
+  );
+}
```

Comment:
anatomy here is more or less a very rough outline. most of the logic that we add will be related to styles, which _could_ end up impacting the underlying markup structure. either way i think its worth updating this during the implementation phase